composer/trainer/trainer.py

```diff
@@ -1106,6 +1106,14 @@ def __init__(
         # compile config for PyTorch 2.0 or higher
         compile_config: Optional[dict[str, Any]] = None,
     ):
+        if deepspeed_config is not None:
+            raise VersionedDeprecationWarning(
+                'The use of DeepSpeed in Composer is deprecated. Composer is tightly integrated with PyTorch FSDP ' +
+                'which provides similar functionality. Please use the `parallelism_config` parameter instead. Please open '
+                + 'a GitHub issue if you need help migrating from DeepSpeed to FSDP.',
+                remove_version='0.28.0',
```

Comment:
```suggestion
                remove_version='1.0.0',
```
We can tighten later

Reply:
We will plan on removing support for training new models using deepspeed in the version list, but will keep support for loading existing deepspeed models until version 1.0.

Reply:
Updated message to be more specific to what will be removed in 0.28